### DISCUSSION TITLE
Display um 180 Grad drehen?

### DISCUSSION THREAD

**Discussion by mcbo007** [beabel/radiowecker#53] [Fragen & Antworten / Q&A] [answered]
Hallo,

danke für das super Projekt.
Ich würde gerne mein Display um 180 Grad drehen, 
dann passt die AZ-Platine besser in mein Gehäuse.

Wenn ich in der "tft_display.ino", Zeile 397
`tft.setRotation(3);`
auf
`tft.setRotation(1);`
setze, ist das Display um 180 Grad gedreht, 
aber der Touch steht scheinbar noch auf dem Kopf.
Was muss ich dafür noch ändern?

Gruß

Marcel

**Comment by beabel** [comment, marked as answer]
ich werde dies mit in der nächsten version integrieren, das sich der touch mit anpasst je nach rotation. 
im develop werde ich es direkt einbauen, bis zu neuen version müsst ihr es selbst einbauen.
alles in der datei tft_display.ino
fügt vor der function 
```
void onTouchClick(TS_Point p) {
```
diesen code ein:

```
TS_Point transformTouch(TS_Point p) {
  // Initialisiere pTransformed mit den gleichen Werten wie p
  TS_Point pTransformed = p;
  
  switch (tft.getRotation()) {
    case 1:
    // Bei Rotation 1 
      pTransformed.x = tft.width() - p.x; 
      pTransformed.y = tft.height() - p.y; 
      break;
    case 3:
      // Keine Änderung notwendig, da pTransformed bereits p ist
      break;
    default:
      // Standardverhalten, keine Änderung notwendig
      break;
  }

  return pTransformed;
}
```

und direkt nach dem start der function 

```
void onTouchClick(TS_Point p) {
```

fügt ihr das ein:

```
  // Zuerst die Touch-Koordinaten transformieren
  p = transformTouch(p);
```

siehe [develop commit](https://github.com/beabel/radiowecker/commit/483c1ff1eaea40888067188f768229c6de020aac)

**Comment by beabel** [comment]
Ich habe es jetzt direkt im main branch veröffentlicht inkl. neuer version 3.0.3

**Comment by mcbo007** [comment]
Danke, funktioniert  👍
